internal/controller/bootstrap/controlplane_bootstrap_controller.go

```diff
@@ -603,12 +603,17 @@ func (c *ControlPlaneController) findFirstControllerIP(ctx context.Context, firs
 			break
 		}
 		if addr.Type == clusterv1.MachineInternalIP {
-			ip := net.ParseIP(addr.Address)
-			if len(ip) == net.IPv4len {
-				intIPv4Addr = ip.To4().String()
+			ip, err := netip.ParseAddr(addr.Address)
+			if err != nil {
+				continue
+			}
+			if ip.Is4() {
+				intIPv4Addr = ip.String()
 				break
 			}
-			intAddr = fmt.Sprintf("[%s]", ip.To16().String())
+			if ip.Is6() {
+				intAddr = ip.WithZone("").String()
```

Comment:
Don't you need the brackets here?

```suggestion
				intAddr = fmt.Sprintf("[%s]", ip.WithZone(""))
```

Reply:
Right, fixed!